tv-shows/src/components/features/shows/ShowReviewForm/ShowReviewForm.tsx

```diff
@@ -1,89 +1,92 @@
+export const ShowReviewForm = () => {
+  const params = useParams();
+  const { mutate } = useSWRConfig();
+  const [loading, setLoading] = useState(false);
 
-  const onClickHandler = () => {
-    const titleValue = titleInputRef.current?.value || "";
-    const commentValue = commentInputRef.current?.value || "";
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset,
+  } = useForm<IReviewDataParams>();
 
-    const ratingElement = document.querySelector(
-      'input[name="rating"]:checked'
-    ) as HTMLInputElement | null;
-    const ratingValue = ratingElement ? parseInt(ratingElement.value, 10) : 0;
+  const { trigger } = useSWRMutation(swrKeys.reviews, createReview, {
+    onSuccess: async (data) => {
+      await mutate(swrKeys.reviewList(params.id as string), data, false);
+      reset();
+      setLoading(false);
+    },
+  });
 
-    if (!titleValue || !commentValue || !ratingValue) {
-      alert("Please enter your review title, comment and rating.");
-      return;
-    }
-
-    const newReview: IReview = {
-      title: titleValue,
-      comment: commentValue,
-      rating: ratingValue,
-    };
-
-    onAdd(newReview);
-
-    // Clear the form inputs
-    if (titleInputRef.current) titleInputRef.current.value = "";
-    if (commentInputRef.current) commentInputRef.current.value = "";
-
-    // Reset the rating input
-    const ratingNoneElement = document.getElementById(
-      "rating-none"
-    ) as HTMLInputElement;
-    if (ratingNoneElement) {
-      ratingNoneElement.checked = true;
-    }
+  const onSubmit = (data: IReviewDataParams) => {
+    setLoading(true);
+    data.show_id = Number(params.id);
+    trigger(data);
   };
 
   return (
     <Flex
+      as="form"
       gap="8"
-      direction="column"
+      flexDirection="column"
       marginBlockEnd="20"
       backgroundColor="gray.300"
       color="white.100"
       borderRadius="lg"
       padding="8"
+      onSubmit={handleSubmit(onSubmit)}
     >
       <Heading as="h3" size="lg">
         Add your review
       </Heading>
 
-      <Input
-        id="title-input"
-        ref={titleInputRef}
-        placeholder="Enter review title"
-        variant="flushed"
-        backgroundColor="gray.100"
-        paddingInline="4"
-      />
-
-      <Textarea
-        id="comment-input"
-        ref={commentInputRef}
-        variant="flushed"
-        placeholder="Enter review comment"
-        backgroundColor="gray.100"
-        paddingInline="4"
-      />
+      <FormControl isInvalid={!!errors.comment}>
+        <FormLabel htmlFor="comment-input">Comment</FormLabel>
+        <Textarea
+          id="comment-input"
+          variant="flushed"
+          placeholder="Enter review comment"
+          backgroundColor="gray.100"
+          paddingInline="4"
+          {...register("comment", { required: "Comment is required" })}
+        />
+        <FormErrorMessage>
+          {errors.comment && errors.comment.message}
```

Comment:
By default je conditional. (zbog onog isInvalid u FormControl)
```suggestion
          {errors.comment.message}
```